### ISSUE TITLE
add brackets for builtin domains where it makes sense

### ISSUE THREAD

**Issue by cos** [kframework/k-legacy#1929]

**Comment by radumereuta** [comment]
@cos, you have to be more descriptive of these issues.
You are not communicating efficiently here.
If this is something just for you to remember, then use something privately, don't clutter everybody's inbox.

**Comment by grosu** [comment]
@radumereuta , no, Cosmin is doing the right thing.  He's adding tasks that need to be completed before Jan 18 as issues, to we can keep track of them and assign them to people.  If anybody is interested in picking up any of these tasks, please volunteer.  Otherwise, most likely you will be assigned to some of these tasks.

**Comment by cos** [comment]
@radumereuta, I'm referencing them back from the pull request -- you can look at the discussion there

**Comment by daejunpark** [comment]
@radumereuta's suggestion doesn't work:

> In `kast.k` line 51, there is this production:
> 
> ```
> syntax KBott     ::= "`" K "`"    [bracket]
> ```
> 
> To add parenthesis as brackets, just add (I would suggest modify and change all the current examples):
> 
> ```
> syntax KBott     ::= "(" K ")"    [bracket]
> ```

I tried the suggestion and tested against kool-untyped, but parsing failed many places (having 82 parsing errors). I guess it is due to ambiguities between an user-defined function's parenthesis with the bracket parenthesis of K.

Note that I tried both ways: either having additional bracket syntax or replacing it, but both failed.

**Comment by radumereuta** [comment]
Hmmm. I will have a look when I get back home tomorrow. It worked in K3. It should work in K4 too.

**Comment by daejunpark** [comment]
I split the bracket for K into separate issue #1943

**Comment by cos** [comment]
@daejunpark, @laurayuwen, is it ok to close this issue?

**Comment by daejunpark** [comment]
yes
